odoo/addons/base/models/ir_actions_report.py

```diff
@@ -678,23 +679,20 @@ def close_streams(streams):
         if len(streams) == 1:
             result = streams[0].getvalue()
         else:
-            try:
-                result = self._merge_pdfs(streams)
-            except utils.PdfReadError:
-                raise UserError(_("One of the documents, you try to merge is encrypted"))
+            result = self._merge_pdfs(streams)
 
         # We have to close the streams after PdfFileWriter's call to write()
         close_streams(streams)
         return result
 
     def _merge_pdfs(self, streams):
-        writer = PdfFileWriter()
+        merged_pdf = Pdf.new()
         for stream in streams:
-            reader = PdfFileReader(stream)
-            writer.appendPagesFromReader(reader)
+            src_pdf = Pdf.open(stream)
+            merged_pdf.pages.extend(src_pdf.pages)
         result_stream = io.BytesIO()
+        merged_pdf.save(result_stream)
```

Comment:
I get the impression that's mostly for if the Pdf object is used to directly open a file handle, which doesn't happen here. But it's a good point, I'll check. I deliberately did not close the `stream` object because there's a `close_streams` call later on: https://github.com/odoo/odoo/blob/14.0/odoo/addons/base/models/ir_actions_report.py#L687 . The way the `_post_pdf` pipeline currently works is all the streams get put into a list and they're all closed together at the end. Closing them in `_merge_pdfs` as a side-effect is maybe a bit awkward. I'm not sure if the `with` context closing the `Pdf.open(stream)` will also close the stream, I'll check.